src/eggCookingHelper.tsx

```diff
@@ -0,0 +1,255 @@
+import { useEffect, useState } from "react";
+
+function createAudio() {
+    const audio = new Audio("/alert.mp3");
+    audio.loop = true;
+    return audio;
+}
+
+export default function EggCookingHelper() {
+    type EggSize = "S" | "M" | "L";
+    type BoilingLevel = "Soft" | "Medium" | "Hard";
+
+    const [eggSize, setEggSize] = useState<EggSize>("M");
+    const [boilingLevel, setBoilingLevel] = useState<BoilingLevel>("Medium");
+    const [remainingTimerSeconds, setRemainingTimerSeconds] = useState(0);
+    const [intervalID, setIntervalID] = useState<number>();
+    const [alertActive, setAlertActive] = useState(false);
+    const [audio] = useState(createAudio());
+
+    const cookingTimes = new Map<EggSize, Map<BoilingLevel, number>>();
```

Comment:
Whenever you define something in a component, consider: "Does this actually need to be defined inside of the component?"  

The reason to always ask yourself this is performance. Because everything that is written inside of the body of the React component (so in its render cycle) will be recalculated on every render.

In this case it does not need to be inside of the component, because it has no dependencies to a state or other thing defined in here. It's just defining the cooking times which is a core data definition.

So you can move this out of the component and do it in the top level of this file instead. Same of course with the setters of the single cooking times.